source/slang/slang-type-layout.cpp

```diff
@@ -398,12 +400,115 @@ struct CPULayoutRulesImpl : DefaultLayoutRulesImpl
     }
 };
 
-// TODO(JS): Most likely wrong. For layout for CUDA, we'll just do the default to get things up and running
 struct CUDALayoutRulesImpl : DefaultLayoutRulesImpl
 {
     typedef DefaultLayoutRulesImpl Super;
-};
 
+    SimpleLayoutInfo GetScalarLayout(BaseType baseType) override
+    {
+        switch (baseType)
+        {
+            case BaseType::Bool:
+            {
+                // In memory a bool is a byte. BUT when in a vector or matrix it will actually be a int32_t
```

Comment:
This comment is a bit confusing here, because it is describing an exception to the rules that comes along later. Reading the comment I had assumed this code was going to return `sizeof(uint32_t)`, so I skimmed past the definition at first, and then only scrolled back later when I saw the code in the vector/matrix cases and thought "so wait, when does it actually get the byte-sized thing?"

Reply:
Also: it might make sense to factor the commonalities between CPU and CUDA layout into a base class, since I'm guessing the two cases overlap a fair bit.